kivy/input/providers/wm_common.py

```diff
@@ -116,14 +117,26 @@ def _event_type(self):
                 return 'end'
         event_type = property(_event_type)
 
+    def SetWindowLong_WndProc_wrapper_generator(func):
+        def _closure(hWnd, wndProc):
+            oldAddr = func(hWnd, GWL_WNDPROC, cast(wndProc, c_void_p).value)
```

Comment:
This function is used both for `SetWindowLongPtrW` and `SetWindowLongW`. But for the latter case, why are you casting to void pointer? I get that on 32 bit a void pointer and a long probably are the same, but still.

Reply:
As I specified in the description, I did not test on *32bit* (I don't want to install all the modules on any of my *32bit* *VEnv*s).

Nevertheless, this is consistent behavior, and more:
- `LONGLONG` and `void*` are the same on *64bit* (8)
- `LONG` and `void*` are the same on *32bit* (4)

Even if their size matches (and from *C* one could get away with a compiler warning - depending on flags), they are still different (from *ctypes* perspective). That's why all those conversions are required.

Bottom line: there's absolutely no difference between *64bit* and *32bit* cases.

Reply:
Yeah, I guess it's not worth replicating the function.